plugins/inventory/inventory.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2022, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+name: inventory
+author:
+  - Polona Mihalič (@PolonaM)
+short_description: Inventory source from Terraform state file.
+description:
+  - Builds an inventory from specified state file.
+  - Does not support caching.
+version_added: 1.0.0
+seealso: []
+options:
+  plugin:
+    description:
+      - The name of the Inventory Plugin.
+      - This should always be C(cloud.terraform.inventory).
+    required: true
+    type: str
+    choices: [ cloud.terraform.inventory ]
+    version_added: 1.0.0
+  project_path:
+    description:
+      - The path to the root of the Terraform directory with the .tfstate file.
+      - If I(state_file) and I(project_path) are not specified, C(terraform.tfstate) file in the current
+        working directory will be used.
+    type: path
+    version_added: 1.0.0
+  state_file:
+    description:
+      - Absolute path to an existing Terraform state file whose outputs will be listed.
+      - If I(state_file) and I(project_path) are not specified, C(terraform.tfstate) file in the current
+        working directory will be used.
+    type: path
+    version_added: 1.0.0
+  binary_path:
+    description:
+      - The path of a terraform binary to use.
+    type: path
+    version_added: 1.0.0
+"""
+
+EXAMPLES = r"""
+# Example that creates an inventory from terraform.tfstate file in cwd
+
+plugin: cloud.terraform.inventory
+
+
+# Example that creates an inventory from mycustomstate.tfstate file in selected project directory
+
+plugin: cloud.terraform.inventory
+project_path: some/project/path
+state_file: mycustomstate.tfstate
```

Comment:
These are mutually exclusive in existing modules, please keep that functionality here too. This requires the description in the documentation to be changed too.

Reply:
They are nowhere mutually exclusive, but to make it more consistent, I would suggest updating terraform_output module

Reply:
As discussed I only added additional examples